src/lib.rs

```diff
@@ -76,6 +76,12 @@ pub trait Logos<'source>: Sized {
     ) -> Lexer<'source, Self> {
         Lexer::with_extras(source, extras)
     }
+
+    /// Create a new error. The default implementation uses `Error::default()`. If you want to make
+    /// your own, use `#[logos(error_callback = ...)]`
+    fn make_error(_lexer: &mut Lexer<'source, Self>) -> Self::Error {
+        Self::Error::default()
+    }
```

Comment:
```suggestion
    #[inline(always)]
    fn make_error(_lexer: &mut Lexer<'source, Self>) -> Self::Error {
        Self::Error::default()
    }
```

We can probably try this to see if that helps the compiler.

Reply:
Performance is unchanged, unfortunately:
```
group                                         default_before                         default_changes
-----                                         --------------                         ---------------
count_ok/identifiers                          1.20   620.0±18.12ns  1198.2 MB/sec    1.00   518.1±28.66ns  1433.8 MB/sec
count_ok/keywords_operators_and_punctators    1.12  1731.6±95.53ns  1173.6 MB/sec    1.00  1552.2±150.95ns  1309.3 MB/sec
count_ok/strings                              1.00    410.6±6.04ns  2023.2 MB/sec    1.40   574.0±12.94ns  1447.1 MB/sec
iterate/identifiers                           1.15   594.5±30.62ns  1249.7 MB/sec    1.00   518.0±27.87ns  1434.3 MB/sec
iterate/keywords_operators_and_punctators     1.04  1623.9±49.04ns  1251.5 MB/sec    1.00  1559.1±111.50ns  1303.5 MB/sec
iterate/strings                               1.00   412.3±32.70ns  2014.8 MB/sec    1.39   571.8±16.91ns  1452.7 MB/sec
```